APMrover2/AP_MotorsUGV.cpp

```diff
@@ -351,14 +351,57 @@ void AP_MotorsUGV::output_throttle(SRV_Channel::Aux_servo_function_t function, f
 }
 
 // slew limit throttle for one iteration
-void AP_MotorsUGV::slew_limit_throttle(float dt)
+void AP_MotorsUGV::slew_limit_throttle_and_steering(float dt)
 {
     if (_slew_rate > 0) {
-        float temp = _slew_rate * dt * 0.01f * (_throttle_max - _throttle_min);
-        if (temp < 1.0f) {
-            temp = 1.0f;
+        // slew throttle
+        const float throttle_change_max = MAX(1.0f, _slew_rate * dt * 0.01f * (_throttle_max - _throttle_min));
+        if (_throttle > _throttle_prev + throttle_change_max) {
+            _throttle = _throttle_prev + throttle_change_max;
+            limit.throttle_upper = true;
+        } else if (_throttle < _throttle_prev - throttle_change_max) {
+            _throttle = _throttle_prev - throttle_change_max;
+            limit.throttle_lower = true;
         }
-        _throttle = constrain_int16(_throttle, _last_throttle - temp, _last_throttle + temp);
+
+        // skid-steering slew steering and handles transition from forward to reverse
+        if (have_skid_steering()) {
+            // calc maximum steering change
+            const float steering_change_max = _slew_rate * dt * 0.01f * 9000.0f;
+            // if throttle has reversed, slew throttle and steering towards zero
+            if (is_negative(_throttle) != is_negative(_throttle_prev)) {
+                if (_steering >= 0) {
+                    _steering = MIN(_steering, _steering_prev);
+                    _steering = MAX(0.0f, _steering - steering_change_max);
+                    limit.steer_right = true;
+                } else {
+                    _steering = MAX(_steering, _steering_prev);
+                    _steering = MIN(0.0f, _steering + steering_change_max);
+                    limit.steer_left = true;
+                }
+                // hold throttle at zero during the transition
+                if (is_zero(_steering)) {
+                    _throttle_prev = _throttle;
+                } else {
+                    _throttle = 0.0f;
```

Comment:
I don't think this is good, that the same thing that is the current throttle controler, but it leave the throttle uncontroled instead of transitionning it to zero. 
Moreover, in skid steering, that means that we get a speed and trigger pivot turn that currently has a discontinuity, one wheel side will switch direction instantanously and risk mechanical breaking !

Reply:
I haven't test this yet ... I need to check